# Week of 2024-06-24 to 2024-07-01 (31)

### GHFirst (9)

- [Revert "[cond] inlining into one of the branches when pred is a python constant (#128709)"](https://github.com/pytorch/pytorch/commit/dfd55d1714cd3968dfb7a6779c0cc4408ce6b583)
  - Sorry for reverting your change but it is breaking one ExecuTorch test ([comment](https://github.com/pytorch/pytorch/pull/128709#issuecomment-2197806850))
- [Revert "[BE][Easy] use `pathlib.Path` instead of `dirname` / `".."` / `pardir` (#129374)"](https://github.com/pytorch/pytorch/commit/3d962178917ccf0fd006423ade30fd1bed659fa9)
  - Sorry for reverting your change but it is still failing with the same error ([comment](https://github.com/pytorch/pytorch/pull/129374#issuecomment-2197801405))
- [Revert "Conversions between strided and jagged layouts for Nested Tensors (#115749)"](https://github.com/pytorch/pytorch/commit/fa6c0fe3e4eca90bbb44ac4e9383799b4cf95eb0)
  - Diff reverted internally ([comment](https://github.com/pytorch/pytorch/pull/115749#issuecomment-2197790226))
- [Revert "[cuDNN][SDPA] Remove `TORCH_CUDNN_SDPA_ENABLED=1`, enable cuDNN SDPA by default on H100 and 2nd on other archs >= sm80 (#125343)"](https://github.com/pytorch/pytorch/commit/999eec8deab9c54343730fedcfefad8fe2932e6d)
  - Sorry for reverting your change but it seems to break some test_transformer running on internal A100 and V100 ([comment](https://github.com/pytorch/pytorch/pull/125343#issuecomment-2196202003))
- [Revert "[nn-module] Use standard dict for _parameters, _modules and _buffers (#129164)"](https://github.com/pytorch/pytorch/commit/8ba0f6c7c2e5547c6f12e664a85bcbdca442511b)
  - Sorry for reverting your change but it seems to break some internal dper3 tests ([comment](https://github.com/pytorch/pytorch/pull/129164#issuecomment-2195888838))
- [Revert "[BE][Easy] use `pathlib.Path` instead of `dirname` / `".."` / `pardir` (#129374)"](https://github.com/pytorch/pytorch/commit/895316119d7971c4d57f7bfbcee30ebd4eff4ed3)
  - Sorry for reverting your change but it causes lots of internal build failures where they fail to find hipify module ([comment](https://github.com/pytorch/pytorch/pull/129374#issuecomment-2192437052))
- [Revert "[halide-backend] Initial implementation of HalideKernel and HalideScheduling (#126417)"](https://github.com/pytorch/pytorch/commit/1a54bb0f9689eb64d92dd676a7aeb7437e3d4b4d)
  - breaking internal builds ([comment](https://github.com/pytorch/pytorch/pull/126417#issuecomment-2186999121))
- [Revert "[halide-backend] Generate standalone runtime (#129025)"](https://github.com/pytorch/pytorch/commit/063facf352a7fd4812b1e87d4f711a71079c6ad3)
  - breaking internal builds ([comment](https://github.com/pytorch/pytorch/pull/129025#issuecomment-2186995467))
- [Revert "[inductor] Refactor fusion of inplace operations (#128979)"](https://github.com/pytorch/pytorch/commit/7b910285db83c77455ee9b5636e8467bfa117d17)
  - breaking internal builds ([comment](https://github.com/pytorch/pytorch/pull/128979#issuecomment-2186846940))

### Ignored Signal (2)

- [Revert "[Traceable FSDP2] Don't decompose fsdp.split_with_sizes_copy (#129414)"](https://github.com/pytorch/pytorch/commit/45b2931b7edb4fc9a0f330b0784d2a388857cfc3)
  - This PR is seems to be causing multiple macos failures.  Looks like it was merged before trunk jobs were started, which would have run those tests ([comment](https://github.com/pytorch/pytorch/pull/129414#issuecomment-2189479505))
- [Revert "[Traceable FSDP2] Add Dynamo support for run_with_rng_state HOP (#127247)"](https://github.com/pytorch/pytorch/commit/fb40ba6fc2af600a5be6fb8040a0a11df5e60220)
  - This PR is seems to be causing multiple macos failures.  Looks like it was merged before trunk jobs were started, which would have run those tests ([comment](https://github.com/pytorch/pytorch/pull/129414#issuecomment-2189479505))

### Landrace (3)

- [Revert "[Inductor] FlexAttention supports block sparse mask (#129216)"](https://github.com/pytorch/pytorch/commit/c43923a11692f71066c18f2f1e230371aa9a353a)
  - Sorry for reverting your change, but it is still failing in trunk https://hud.pytorch.org/pytorch/pytorch/commit/b9d3cedd648d4ed9d0bf5b918893341e5f95289a, maybe a landrace given that TD has been turned off ([comment](https://github.com/pytorch/pytorch/pull/129216#issuecomment-2196182882))
- [Revert "[AOTI][refactor] Unify UserDefinedTritonKernel.codegen (#129378)"](https://github.com/pytorch/pytorch/commit/f7708ffebb4557383904f2ddefd5f596229e0725)
  - broke inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCuda::test_triton_kernel_sympy_expr_arg_abi_compatible_cuda and a few other tests https://github.com/pytorch/pytorch/actions/runs/9680978494/job/26713689249 https://hud.pytorch.org/pytorch/pytorch/commit/52009068bc39ebc846bd37b44f5f9c5f62257778. The tests were added in https://github.com/pytorch/pytorch/pull/129301 which is before your base ([comment](https://github.com/pytorch/pytorch/pull/129378#issuecomment-2192032697))
- [Revert "[FSDP2] Ran post-acc-grad hooks manually (#129450)"](https://github.com/pytorch/pytorch/commit/52341c28e817ee6bc36b529823f8248ba395d5bb)
  - broke distributed/_composable/fsdp/test_fully_shard_compile.py::TestFullyShardCompile::test_simple_mlp_fullgraph_backend_aot_eager https://hud.pytorch.org/pytorch/pytorch/commit/7ebffef4d02a3cc68dbbcf44b92d63c7fe0ebb67 https://github.com/pytorch/pytorch/actions/runs/9667812641/job/26671489454.  Test got added in https://github.com/pytorch/pytorch/pull/129157 which is before your mergebase ([comment](https://github.com/pytorch/pytorch/pull/129450#issuecomment-2190174363))

### Not through pytorchbot (1)

- [Revert "[LLVM/TensorExpr] Update for an API change in LLVM 18." (#129797)](https://github.com/pytorch/pytorch/commit/89696db4b05759bf0db271bd0c256b42b31df9b5)

### No Signal (4)

- [Revert "[BE][Easy] replace `import pathlib` with `from pathlib import Path` (#129426)"](https://github.com/pytorch/pytorch/commit/2effbcfcd8a3dc63e803d6729fb3e4f750841dff)
  - recognize `Path` as new exported API ([comment](https://github.com/pytorch/pytorch/pull/129426#issuecomment-2198371625))
- [Revert "Add warning for weights_only (#129239)"](https://github.com/pytorch/pytorch/commit/b1f486aff9f3ae2f0cec557fe2ce5005e1a192bb)
  - Sorry for reverting your change but I am seeing some test_nn failures from ROCm https://hud.pytorch.org/pytorch/pytorch/commit/381ce0821c3fa2b342f0b8660c76cc27f48543c4, trying to revert this to see if trunk recovers ([comment](https://github.com/pytorch/pytorch/pull/129239#issuecomment-2189812903))
- [Revert "Add example for torch.serialization.add_safe_globals (#129396)"](https://github.com/pytorch/pytorch/commit/7cf454ec52e2ee431faa365a2998720047bc970a)
  - Sorry for reverting your change but I am seeing some test_nn failures from ROCm https://hud.pytorch.org/pytorch/pytorch/commit/381ce0821c3fa2b342f0b8660c76cc27f48543c4, trying to revert this to see if trunk recovers ([comment](https://github.com/pytorch/pytorch/pull/129239#issuecomment-2189812903))
- [Revert "[BE] update type annotations for basic utilities in `torch/__init__.py` (#129001)"](https://github.com/pytorch/pytorch/commit/cb4919344a62325cf7c33d703d041a048118659b)
  - lint failure ([comment](https://github.com/pytorch/pytorch/pull/129001#issuecomment-2186944549))

### Weird (12)

- [Revert "[BE][Easy] enable postponed annotations in `tools` (#129375)"](https://github.com/pytorch/pytorch/commit/a32ce5ce344b384ad4c6642895835c10ae7872c8)
  - Sorry for reverting your change but I need to revert to cleanly revert https://github.com/pytorch/pytorch/pull/129374, please do a rebase and reland this ([comment](https://github.com/pytorch/pytorch/pull/129375#issuecomment-2197800541))
- [Revert "[BE][Easy] enable postponed annotations in `torchgen` (#129376)"](https://github.com/pytorch/pytorch/commit/6063bb9d45bcc1439f00f7c6fffa3975f9b35606)
  - Sorry for reverting your change but I need to revert to cleanly revert https://github.com/pytorch/pytorch/pull/129374, please do a rebase and reland this ([comment](https://github.com/pytorch/pytorch/pull/129375#issuecomment-2197800541))
- [Revert "Use Generic TypeAlias (PEP 585) and Union Type (PEP 604) in `.pyi` stub files (#129419)"](https://github.com/pytorch/pytorch/commit/83caf4960f36d50032a6945fc2124a76af086ce0)
  - Sorry for reverting your change but I need to revert to cleanly revert https://github.com/pytorch/pytorch/pull/129374, please do a rebase and reland this ([comment](https://github.com/pytorch/pytorch/pull/129375#issuecomment-2197800541))
- [Revert "[BE] enforce style for empty lines in import segments (#129751)"](https://github.com/pytorch/pytorch/commit/00d7bba2fa8c1f68f06c4f07c1b0875c11afb439)
  - Sorry for reverting your change but I need to revert to cleanly revert https://github.com/pytorch/pytorch/pull/129374, please do a rebase and reland this ([comment](https://github.com/pytorch/pytorch/pull/129751#issuecomment-2197799814))
- [Revert "[cuDNN][SDPA] Bail out of dispatching to cuDNN for head dim > 128 on Ampere (#129587)"](https://github.com/pytorch/pytorch/commit/d21993bbb8b9ed1a534a594601fb6065c1d116ec)
  - Sorry for revert yet another of your change but I need to revert this to cleanly revert https://github.com/pytorch/pytorch/pull/125343#issuecomment-2196187332 ([comment](https://github.com/pytorch/pytorch/pull/129587#issuecomment-2196198756))
- [Revert "[Inductor] FlexAttention supports block sparse mask (#129216)"](https://github.com/pytorch/pytorch/commit/5ceba6a3cbd817922af0363f6ba39a85ab58e309)
  - broke functorch/aot_dispatch and test_proxy_tensor on windows https://github.com/pytorch/pytorch/actions/runs/9691331440/job/26743164471 https://hud.pytorch.org/pytorch/pytorch/commit/4082759925a712b7cb340164d3da3a1dab372d9f missed on PR due to bad TD ([comment](https://github.com/pytorch/pytorch/pull/129216#issuecomment-2195087274))
- [Revert "[CUDA][Inductor][CI] Revert PR#127150 since cu124 is now behaving similar enough to cu121 (#128423)"](https://github.com/pytorch/pytorch/commit/e9aefad641f24072f36badf4b221ebca3f222966)
  - Sorry for reverting your change but I need to revert it to cleanly revert https://github.com/pytorch/pytorch/pull/129374 ([comment](https://github.com/pytorch/pytorch/pull/128423#issuecomment-2192423840))
- [Revert "[ALI] [Reland] Use LF runners for Lint (#129071)"](https://github.com/pytorch/pytorch/commit/211f38e74207ab9070a78b8d699f0e3967249eca)
  - All LF jobs are backlogged, so revert this one ([comment](https://github.com/pytorch/pytorch/pull/129071#issuecomment-2191676677))
- [Revert "Remove test_mps_allocator_module XFAIL (#129340)"](https://github.com/pytorch/pytorch/commit/b045878f81d797044a1e9af5bf2caf3f0b2e704c)
  - The test is now failing again in trunk after a day or so of staying green, we need to continue the investigation ([comment](https://github.com/pytorch/pytorch/pull/129340#issuecomment-2189701706))
- [Revert "[inductor] Fix TORCHINDUCTOR_FORCE_DISABLE_CACHES (#129257)"](https://github.com/pytorch/pytorch/commit/ad76da6c16c5dc465e8aac8d913532251db7b400)
  - one of the PRs in the stack seems to have broken test/distributed/_composable/test_replicate_with_compiler.py::ReplicateTest::test_bucketing_concat_op on distributed https://github.com/pytorch/pytorch/actions/runs/9653941844/job/26627760340 https://hud.pytorch.org/pytorch/pytorch/commit/4c1e4c5f307f9743014a08cf97d3fa8de7e1ce5f, not tested on this PR due to bad TD ([comment](https://github.com/pytorch/pytorch/pull/129257#issuecomment-2189444171))
- [Revert "[inductor] Enable FX graph caching in OSS by default (#125863)"](https://github.com/pytorch/pytorch/commit/b38f6d4cd23fd3433361b9b06cb46726be9b658c)
  - one of the PRs in the stack seems to have broken test/distributed/_composable/test_replicate_with_compiler.py::ReplicateTest::test_bucketing_concat_op on distributed https://github.com/pytorch/pytorch/actions/runs/9653941844/job/26627760340 https://hud.pytorch.org/pytorch/pytorch/commit/4c1e4c5f307f9743014a08cf97d3fa8de7e1ce5f, not tested on this PR due to bad TD ([comment](https://github.com/pytorch/pytorch/pull/129257#issuecomment-2189444171))
- [Revert "Add Strided Input test for flex attention (#128915)"](https://github.com/pytorch/pytorch/commit/c012013aa6c1dc84b1f566dadfce3d7c5217b860)
  - Sorry for reverting your change but its tests are failing in trunk, i.e. https://hud.pytorch.org/pytorch/pytorch/commit/41bb81b58279f492e72bd270b3b071dd2953ed8c#26627138290 ([comment](https://github.com/pytorch/pytorch/pull/128915#issuecomment-2187695317))
